openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TypeRepository.java

```diff
@@ -443,9 +446,56 @@ private void validatePropertyConfigUpdate(
         HashSet<String> updatedValues = new HashSet<>(updatedConfig.getValues());
         if (updatedValues.size() != updatedConfig.getValues().size()) {
           throw new IllegalArgumentException("Enum Custom Property values cannot have duplicates.");
-        } else if (!updatedValues.containsAll(origConfig.getValues())) {
-          throw new IllegalArgumentException(
-              "Existing Enum Custom Property values cannot be removed.");
+        }
+      }
+    }
+
+    private void postUpdateCustomPropertyConfig(
+        Type entity, CustomProperty origProperty, CustomProperty updatedProperty) {
+      if (origProperty.getPropertyType().getName().equals("enum")) {
+        EnumConfig origConfig =
+            JsonUtils.convertValue(
+                origProperty.getCustomPropertyConfig().getConfig(), EnumConfig.class);
+        EnumConfig updatedConfig =
+            JsonUtils.convertValue(
+                updatedProperty.getCustomPropertyConfig().getConfig(), EnumConfig.class);
+        HashSet<String> updatedKeys = new HashSet<>(updatedConfig.getValues());
+
+        if (!updatedKeys.containsAll(origConfig.getValues())) {
+          // Update the extension values with removed enum keys
+          List<String> removedEnumKeys =
+              origConfig.getValues().stream()
+                  .filter(value -> !updatedConfig.getValues().contains(value))
+                  .toList();
+
+          String customPropertyFQN =
+              getCustomPropertyFQN(entity.getName(), updatedProperty.getName());
+
+          List<CollectionDAO.ExtensionWithIdAndSchemaObject> extensions =
+              daoCollection.entityExtensionDAO().getExtensionsByPrefixBatch(customPropertyFQN);
+          List<CollectionDAO.ExtensionWithIdAndSchemaObject> updatedExtensions =
```

Comment:
@sonika-shah this could be a huge update no? depending on no.of tables this property can exist?